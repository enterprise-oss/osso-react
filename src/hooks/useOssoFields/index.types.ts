export { OssoProvider, ProviderMap, Providers } from '../../types';
