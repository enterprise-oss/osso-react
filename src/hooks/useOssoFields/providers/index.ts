import azure from './azure';
import genericSaml from './genericSaml';
import google from './google';
import okta from './okta';
import oneLogin from './oneLogin';
import ping from './ping';
import salesforce from './salesforce';

export { azure, genericSaml, google, okta, oneLogin, ping, salesforce };
