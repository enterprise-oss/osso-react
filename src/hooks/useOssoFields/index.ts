import { OssoProvider, ProviderMap, ProviderOption, Providers } from './index.types';
import * as allProviders from './providers';

const providerMap = Object.values(allProviders).reduce((map, provider) => {
  return { ...map, [provider.value as Providers]: provider };
}, {} as ProviderMap<Providers>);

const providers: ProviderOption[] = Object.values(allProviders).map((provider) => ({
  value: provider.value,
  icon: provider.icon,
  label: provider.label,
}));

const useOssoFields = (): {
  fieldsForProvider: (provider: Providers) => OssoProvider;
  providers: ProviderOption[];
} => {
  return {
    providers,
    fieldsForProvider: (provider: Providers) => providerMap[provider],
  };
};

export default useOssoFields;
