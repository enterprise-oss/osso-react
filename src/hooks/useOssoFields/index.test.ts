import { renderHook } from '@testing-library/react-hooks';

import useOssoFields from './index';

describe('providers', () => {
  test('contains an array of Identity Providers in the tuple', () => {
    const {
      result: {
        current: { providers },
      },
    } = renderHook(() => useOssoFields());

    expect(providers).toMatchSnapshot();
  });

  test('each Identity Provider has a label, value, and icon', () => {
    const {
      result: {
        current: { providers },
      },
    } = renderHook(() => useOssoFields());

    providers.map((provider) => {
      expect(Object.keys(provider).sort).toEqual(['value', 'label', 'icon'].sort);
    });
  });
});

describe('fieldsForProvider', () => {
  test('it returns the expected fields for each provider', () => {
    const {
      result: {
        current: { providers, fieldsForProvider },
      },
    } = renderHook(() => useOssoFields());

    providers.map((providerOption) => {
      const result = fieldsForProvider(providerOption.value);
      expect(result).toMatchSnapshot();
    });
  });
});
