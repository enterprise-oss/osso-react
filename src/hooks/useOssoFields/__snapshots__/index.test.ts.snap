// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fieldsForProvider it returns the expected fields for each provider 1`] = `
Object {
  "icon": "test-file-stub",
  "idpGeneratedFields": Object {
    "manual": Array [
      Object {
        "inputProps": Object {
          "copyable": false,
          "id": "osso-azure-sso-url",
          "label": "Identity Provider Single Sign-On URL",
          "readOnly": false,
          "type": "text",
        },
        "name": "ssoUrl",
      },
      Object {
        "inputProps": Object {
          "copyable": false,
          "id": "osso-azure-sso-cert",
          "label": "X.509 Certificate",
          "readOnly": false,
          "type": "textarea",
        },
        "name": "ssoCert",
      },
    ],
    "metadataXml": Object {
      "accept": ".xml",
      "copyable": false,
      "id": "osso-azure-metadata-xml",
      "label": "Metadata XML",
      "readOnly": false,
      "type": "textarea",
    },
  },
  "label": "Azure",
  "ossoGeneratedFields": Object {
    "appDetails": Array [
      Object {
        "name": "contactEmail",
        "pdfProps": Object {
          "x": 45,
          "y": 388,
        },
      },
      Object {
        "name": "name",
        "pdfProps": Object {
          "x": 395,
          "y": 962,
        },
      },
    ],
    "manual": Array [
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-azure-acs-url",
          "label": "Reply URL (Assertion Consumer Service URL)",
          "readOnly": true,
          "type": "text",
        },
        "name": "acsUrl",
        "pdfProps": Object {
          "size": 8,
          "x": 55,
          "y": 2880,
        },
      },
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-azure-entity-id",
          "label": "Identifier (Entity ID)",
          "readOnly": true,
          "type": "text",
        },
        "name": "domain",
        "pdfProps": Object {
          "x": 55,
          "y": 2828,
        },
      },
    ],
  },
  "serviceProviderMetadata": false,
  "value": "AZURE",
}
`;

exports[`fieldsForProvider it returns the expected fields for each provider 2`] = `
Object {
  "icon": "test-file-stub",
  "idpGeneratedFields": Object {
    "manual": Array [
      Object {
        "inputProps": Object {
          "copyable": false,
          "id": "osso-okta-sso-url",
          "label": "Identity Provider Single Sign-On URL",
          "readOnly": false,
          "type": "text",
        },
        "name": "ssoUrl",
      },
      Object {
        "inputProps": Object {
          "copyable": false,
          "id": "osso-okta-sso-cert",
          "label": "X.509 Certificate",
          "readOnly": false,
          "type": "textarea",
        },
        "name": "ssoCert",
      },
    ],
    "metadataXml": Object {
      "accept": ".xml",
      "copyable": false,
      "id": "osso-okta-metadata-xml",
      "label": "Metadata XML",
      "readOnly": false,
      "type": "textarea",
    },
  },
  "label": "Okta",
  "ossoGeneratedFields": Object {
    "appDetails": Array [
      Object {
        "name": "contactEmail",
        "pdfProps": Object {
          "x": 45,
          "y": 388,
        },
      },
      Object {
        "name": "name",
        "pdfProps": Object {
          "x": 215,
          "y": 1615,
        },
      },
      Object {
        "name": "logoUrl",
        "pdfProps": Object {
          "x": 81,
          "y": 1684,
        },
      },
    ],
    "manual": Array [
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-okta-acs-url",
          "label": "Single sign on URL",
          "readOnly": true,
          "type": "text",
        },
        "name": "acsUrl",
        "pdfProps": Object {
          "size": 8,
          "x": 55,
          "y": 2009,
        },
      },
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-okta-entity-id",
          "label": "Audience URI (SP Entity ID)",
          "readOnly": true,
          "type": "text",
        },
        "name": "domain",
        "pdfProps": Object {
          "x": 55,
          "y": 2167,
        },
      },
    ],
  },
  "serviceProviderMetadata": false,
  "value": "OKTA",
}
`;

exports[`fieldsForProvider it returns the expected fields for each provider 3`] = `
Object {
  "icon": "test-file-stub",
  "idpGeneratedFields": Object {
    "manual": Array [
      Object {
        "inputProps": Object {
          "copyable": false,
          "id": "osso-one-login-sso-url",
          "label": "Identity Provider Single Sign-On URL",
          "readOnly": false,
          "type": "text",
        },
        "name": "ssoUrl",
      },
      Object {
        "inputProps": Object {
          "copyable": false,
          "id": "osso-one-login-sso-cert",
          "label": "X.509 Certificate",
          "readOnly": false,
          "type": "textarea",
        },
        "name": "ssoCert",
      },
    ],
    "metadataXml": Object {
      "accept": ".xml",
      "copyable": false,
      "id": "osso-one-login-metadata-xml",
      "label": "Metadata XML",
      "readOnly": false,
      "type": "file",
    },
  },
  "label": "OneLogin",
  "ossoGeneratedFields": Object {
    "appDetails": Array [
      Object {
        "name": "contactEmail",
        "pdfProps": Object {
          "x": 42,
          "y": 389,
        },
      },
      Object {
        "name": "name",
        "pdfProps": Object {
          "x": 326,
          "y": 1044,
        },
      },
      Object {
        "name": "logoUrl",
        "pdfProps": Object {
          "x": 81,
          "y": 1164,
        },
      },
    ],
    "manual": Array [
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-one-login-acs-url",
          "label": "Single sign on URL",
          "readOnly": true,
          "type": "text",
        },
        "name": "acsUrl",
        "pdfProps": Object {
          "size": 8,
          "x": 56,
          "y": 1713,
        },
      },
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-one-login-entity-id",
          "label": "Audience URI (SP Entity ID)",
          "readOnly": true,
          "type": "text",
        },
        "name": "domain",
        "pdfProps": Object {
          "x": 56,
          "y": 1554,
        },
      },
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-one-login-acs-url-validator",
          "label": "ACS URL Validator",
          "readOnly": true,
          "type": "text",
        },
        "name": "acsUrlValidator",
        "pdfProps": Object {
          "size": 8,
          "x": 56,
          "y": 1660,
        },
      },
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-one-login-recipient",
          "label": "Recipient",
          "readOnly": true,
          "type": "text",
        },
        "name": "acsUrl",
        "pdfProps": Object {
          "size": 8,
          "x": 56,
          "y": 1607,
        },
      },
    ],
  },
  "serviceProviderMetadata": false,
  "value": "ONELOGIN",
}
`;

exports[`fieldsForProvider it returns the expected fields for each provider 4`] = `
Object {
  "icon": "test-file-stub",
  "idpGeneratedFields": Object {
    "manual": Array [
      Object {
        "inputProps": Object {
          "copyable": false,
          "id": "osso-google-sso-url",
          "label": "Identity Provider Single Sign-On URL",
          "readOnly": false,
          "type": "text",
        },
        "name": "ssoUrl",
      },
      Object {
        "inputProps": Object {
          "copyable": false,
          "id": "osso-google-sso-cert",
          "label": "X.509 Certificate",
          "readOnly": false,
          "type": "textarea",
        },
        "name": "ssoCert",
      },
    ],
    "metadataXml": Object {
      "accept": ".xml",
      "copyable": false,
      "id": "osso-google-metadata-xml",
      "label": "Metadata XML",
      "readOnly": false,
      "type": "textarea",
    },
  },
  "label": "Google (SAML)",
  "ossoGeneratedFields": Object {
    "appDetails": Array [
      Object {
        "name": "contactEmail",
        "pdfProps": Object {
          "x": 40,
          "y": 323,
        },
      },
      Object {
        "name": "name",
        "pdfProps": Object {
          "x": 354,
          "y": 783,
        },
      },
      Object {
        "name": "logoUrl",
        "pdfProps": Object {
          "x": 81,
          "y": 891,
        },
      },
    ],
    "manual": Array [
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-google-acs-url",
          "label": "Single sign on URL",
          "readOnly": true,
          "type": "text",
        },
        "name": "acsUrl",
        "pdfProps": Object {
          "size": 8,
          "x": 56,
          "y": 996,
        },
      },
      Object {
        "inputProps": Object {
          "copyable": true,
          "id": "osso-google-entity-id",
          "label": "Audience URI (SP Entity ID)",
          "readOnly": true,
          "type": "text",
        },
        "name": "domain",
        "pdfProps": Object {
          "x": 56,
          "y": 1049,
        },
      },
    ],
  },
  "serviceProviderMetadata": false,
  "value": "GOOGLE",
}
`;

exports[`providers contains an array of Identity Providers in the tuple 1`] = `
Array [
  Object {
    "icon": "test-file-stub",
    "label": "Azure",
    "value": "AZURE",
  },
  Object {
    "icon": "test-file-stub",
    "label": "Okta",
    "value": "OKTA",
  },
  Object {
    "icon": "test-file-stub",
    "label": "OneLogin",
    "value": "ONELOGIN",
  },
  Object {
    "icon": "test-file-stub",
    "label": "Google (SAML)",
    "value": "GOOGLE",
  },
]
`;
